### ISSUE TITLE
`MayaDT.from_struct` parses incorrectly

### ISSUE THREAD

**Issue by ajfriend** [kennethreitz/maya#104]
There seems to be a problem with the `MayaDT.from_struct`, where it produces an incorrect date. Here's an example of processing the same epoch time two different ways, which gets two different results. 

The block
```python
import time
import maya
import datetime as DT

e = 1507756331

# correct
t = DT.datetime.utcfromtimestamp(e)
t = maya.MayaDT.from_datetime(t)
print t
```
prints `Wed, 11 Oct 2017 21:12:11 GMT`, while the block

```python
# incorrect
t = time.gmtime(e)
t = maya.MayaDT.from_struct(t)
print t
```
prints (incorrectly) `Thu, 12 Oct 2017 05:12:11 GMT`.

**Comment by timofurrer** [comment]
Was solved with #119 .
Thanks for the bug report :tada:

**Comment by vlcinsky** [comment]
either a test case or the code is broken:

```
    def test_issue_104():
        e = 1507756331
        t = Datetime.utcfromtimestamp(e)
        t = maya.MayaDT.from_datetime(t)
        assert str(t) == 'Wed, 11 Oct 2017 21:12:11 GMT'
        t = time.gmtime(e)
        t = maya.MayaDT.from_struct(t)
>       assert str(t) == 'Wed, 11 Oct 2017 21:12:11 GMT'
E       AssertionError: assert 'Wed, 11 Oct ... 22:12:11 GMT' == 'Wed, 11 Oct 2017 21:12:11 GMT'
E         - Wed, 11 Oct 2017 22:12:11 GMT
E         ?                   ^
E         + Wed, 11 Oct 2017 21:12:11 GMT
E         ?                   ^

tests/test_maya.py:80: AssertionError
=================================================
```

Tested on commit:

```
commit 15a5c9eedb52886fc8f59bc31d608066bedb11e4
Merge: b411d5e 7c5d587
Author: Timo Furrer <tuxtimo@gmail.com>
Date:   Fri Mar 30 11:19:14 2018 +0200
```

**Comment by timofurrer** [comment]
The problem seems to be that `time.gmtime(1507756331)` is not in DST, however `MayaDT.from_struct()` uses the current time (which is currently in DST) to calculate the UTC offset. Thus, there's a mismatch between DST. 

I'll have a look at it ;)

**Comment by vlcinsky** [comment]
My timezone is CEST (Central European Summer Time, thus +02:00)

I will try creating a test changing timezones using [freezegun](https://github.com/spulec/freezegun).

**Comment by timofurrer** [comment]
The problem was because of DST not because of your timezone. I've already fixed it with 5f6b5fc

I'll have a look at your test though.
